lib/validations.js

```diff
@@ -340,7 +340,20 @@ function validateCustom(attr, conf, err, options, done) {
     done = options;
     options = {};
   }
-  conf.customValidator.call(this, err, done);
+
+  if (conf.customValidator.length <= 1) {
+    conf.customValidator.call(this, options)
+      .then((isValid) => {
+        if (!isValid) throw new Error();
+        done();
+      })
+      .catch(() => {
+        err();
+        done();
+      });
+  } else {
+    conf.customValidator.call(this, err, done, options);
```

Comment:
I think passing the options after the `done` callback is rather unusual, since the Node.js convention is to pass the callback function as the last arg.

I am proposing the following change:

```js
if (conf.customValidator.length !== 4) {
  // backwards compatibility with custom validators not accepting options
  conf.customValidator.call(this, err, done);
} else {
  conf.customValidator.call(this, options, err, done);
}
```

Thoughts?